### ISSUE TITLE
Navigation using new Authenticator UI component is not working

### ISSUE THREAD

**Issue by andrewcrayden** [aws-amplify/amplify-flutter#1238]
**Describe the bug**
[As suggested in the documentation](https://docs.amplify.aws/lib/auth/getting-started/q/platform/flutter/#authentication-with-amplify), "the Authenticator component should be a descendant of your MaterialApp". The issue is that this Amplify UI Component does not seem to work with the `MaterialApp` navigation system, and there doesn't seem to be any guidance for navigation in the following documentation:
- [Authenticator Amplify UI documentation](https://docs.amplify.aws/ui/auth/authenticator/q/framework/flutter/)
- [Flutter Authentication library documentation](https://docs.amplify.aws/lib/auth/getting-started/q/platform/flutter/#authentication-with-amplify)
- [amplify_authenticator pub.dev Readme and Example pages](https://pub.dev/packages/amplify_authenticator)

The `HomePage` component below renders fine, but the `SecondPage` component which includes the new `SignOutButton` throws the following error:  
> The following assertion was thrown building SignOutButton(dirty): No InheritedAuthViewModel widget found.
Make sure your app is wrapped with an Authenticator widget. The relevant error-causing widget was: SignOutButton

Is this new Authenticator UI component meant to work when configuring the `routes` property of `MaterialApp`? What is the correct way to implement navigation and the new Authenticator UI component together?

**main.dart**
```
import 'package:amplify_api/amplify_api.dart';
import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
import 'package:amplify_authenticator/amplify_authenticator.dart';
import 'package:amplify_flutter/amplify.dart';
import 'package:flutter/material.dart';
import 'pages/home_page.dart';
import 'pages/second_page.dart';
import 'package:example/providers/user_provider.dart';
import 'package:provider/provider.dart';
import 'dart:developer' as developer;
import 'amplifyconfiguration.dart';

void main() {
  runApp(
      MultiProvider(providers: [
        ChangeNotifierProvider(create: (_) => UserProvider())
      ],
          child: MyApp()
      )
  );
}

class MyApp extends StatefulWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {

  @override
  void initState() {
    super.initState();
    _configureAmplify();
  }

  Future<void> _configureAmplify() async {

    AmplifyAuthCognito auth = AmplifyAuthCognito();
    AmplifyAPI api = AmplifyAPI();
    await Amplify.addPlugins([auth, api]);
    try {
      await Amplify.configure(amplifyconfig);
    } on AmplifyAlreadyConfiguredException {
      print("Amplify was already configured. Looks like app restarted on android.");
    } catch(e) {
      developer.log('YOUR ERRROR: ', error: e);
    }
  }

  @override
  Widget build(BuildContext context) {
    final authenticator = Authenticator(
      child: const HomePage(title: 'Home'),
    );

    return MaterialApp(
      title: 'Authenticator Test',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: authenticator,
      routes: {
        '/home': (context) => const HomePage(title: 'Home'),
        '/second-page': (context) => const SecondPage(title: 'Second page'),
      },
    );
  }
}
```

**home_page.dart**

```
import 'package:amplify_authenticator/amplify_authenticator.dart';
import 'package:amplify_flutter/amplify.dart';
import 'package:flutter/material.dart';

class HomePage extends StatefulWidget {
  const HomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {

  @override
  initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
          title: Text(widget.title),
          actions: const <Widget> [
            SignOutButton()
          ]
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            ElevatedButton(
              onPressed: () {Navigator.pushNamed(context, '/second-page');},
              child: const Text('Second page'),
            ),
          ],
        ),
      ),
    );
  }
}
```

**second_page.dart**
```
import 'package:amplify_authenticator/amplify_authenticator.dart';
import 'package:flutter/material.dart';

class SecondPage extends StatefulWidget {
  const SecondPage({Key? key, required this.title}) : super(key: key);
  final String title;
  @override
  _SecondPageState createState() => _SecondPageState();
}

class _SecondPageState extends State<SecondPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
          title: Text(widget.title),
          actions: const <Widget> [
            SignOutButton()
          ]
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'Second page content',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
    );
  }
}
```

**To Reproduce**
Hoping the description and code above is enough to describe the problem. Let me know if more detail is required.

**Expected behavior**
Any page other than home renders correctly, even if there are Authenticator based components such as `SignOutButton`.

**Screenshots**
Happy to provide screenshots if required.

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[] iOS

<details>
Don't think this is required for this issue, but let me know.
  <summary>Output of <code>flutter doctor -v</code></summary>
  
  ```Paste your output of "flutter doctor -v" here```

</details>

<details>
Don't think this is required for this issue, but let me know.
  <summary>Dependencies (<code>pubspec.lock</code>)</summary>
  
  ```
  Paste the contents of your "pubspec.lock" file here
  ```

</details>

**Smartphone (please complete the following information):**
- sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)

**Additional context**
None at this point.

**Comment by dnys1** [comment]
Hi @andrewcrayden - thanks for bringing this to our attention. Can you try the following while we explore better strategies? Basically, it requires creating your navigator as a descendant of the Authenticator.

```dart
final routes = {
  '/routeA': (BuildContext context) => const RouteA(),
  '/routeB': (BuildContext context) => const RouteB(),
};

final authenticator = Authenticator(
  child: Navigator(
    initialRoute: '/routeA',
    onGenerateRoute: (settings) {
      final builder = routes[settings.name];
      if (builder == null) {
        return null;
      }
      return MaterialPageRoute<dynamic>(builder: builder);
    },
  ),
);

return MaterialApp(
  title: 'Authenticator Demo',
  theme: ThemeData.light(),
  darkTheme: ThemeData.dark(),
  themeMode: ThemeMode.system,
  debugShowCheckedModeBanner: false,
  home: authenticator,
);
```

**Comment by andrewcrayden** [comment]
Hi @dnys1, thanks for the quick reply! The solution you have provided works on my end. Looking forward to seeing what navigation strategies are introduced to this UI component as it evolves.

**Comment by dnys1** [comment]
My pleasure. Thanks for trying the Authenticator! I think we've found a better way to do it and can hopefully release that soon for you and others to try.

**Comment by dnys1** [comment]
@andrewcrayden We've released the updated version: `0.1.0-rc.2`. Check out the docs again to see the changes required (pretty minor): https://docs.amplify.aws/ui/auth/authenticator/q/framework/flutter/#basic-integration
